client/lib/sites-list/sites-observer.jsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import omit from 'lodash/omit';
+
+/**
+ * Internal dependencies
+ */
+import Debug from 'debug';
+const debug = Debug( 'calypso:site-observe-hoc' );
+
+export const sitesObserver = ( WrappedComponent ) => {
+	class SitesObserverComponent extends Component {
+
+		constructor( props, context ) {
+			super( props, context );
+
+			this.update = this.update.bind( this );
+
+			this.state = {};
+		}
+
+		componentDidMount() {
+			this.props.sites.on( 'change', this.update );
+			this.cacheSites();
+		}
+
+		componentWillUnmount() {
+			this.props.sites.off( 'change', this.update );
+		}
+
+		update() {
+			debug( 'Re-rendering ' + this.constructor.displayName + ' component.' );
+			this.cacheSites();
+		}
+
+		cacheSites() {
+			if ( this.props.sites ) {
+				this.setState( { sites: Object.create( this.props.sites ) } );
```

Comment:
The reason is that we need to create a new object instance so that React shallow-compare will detect that a change has been made. `Objcet.create` seem like the best solution as it creates a new object that is effectively the same as `this.props.sites` which is the mutable `SitesList` instance.

I've checked how `this.props.sites` is used in the downstream components, to my best judgment, it doesn't seem like it will cause any problems down the road.

Do you see any problems it might cause?
Do you have alternatives to `Object.create`?